packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -465,11 +469,38 @@ export function requestUpdateLane(fiber: Fiber): Lane {
     const currentLanePriority = getCurrentUpdateLanePriority();
     lane = findUpdateLane(currentLanePriority, currentEventWipLanes);
   } else {
-    const schedulerLanePriority = schedulerPriorityToLanePriority(
-      schedulerPriority,
-    );
-
-    lane = findUpdateLane(schedulerLanePriority, currentEventWipLanes);
+    let eventPriority = DefaultEvent;
+    if (enableNativeEventPriorityInference) {
+      eventPriority = getCurrentEventPriority();
+    }
+    switch (eventPriority) {
+      case DiscreteEvent: {
+        lane = findUpdateLane(InputDiscreteLanePriority, currentEventWipLanes);
+        break;
+      }
+      case ContinuousEvent: {
+        lane = findUpdateLane(
+          InputContinuousLanePriority,
+          currentEventWipLanes,
+        );
+        break;
+      }
+      default: {
+        if (__DEV__) {
+          console.error(
+            'Invalid event priority returned from getCurrentEventPriority: %s.',
+            eventPriority,
+          );
+        }
+      }
+      // eslint-disable-next-line no-fallthrough
+      case DefaultEvent: {
+        const schedulerLanePriority = schedulerPriorityToLanePriority(
```

Comment:
This can be in a follow-up, but let's move this Scheduler consultation to React DOM. We only need to check it for message events, since that's the event Scheduler uses internally.

Conceptually, we're asking the renderer "what is the priority of the current platform event?" Scheduler isn't part of the The Platform but it's standing in for `postTask` which is. So one day we can swap it out for that.

And React Native will probably do something completely different, so makes sense to move this logic to the renderer.

Reply:
Currently, this can return Idle or NoLane priorities. Are these also valid? E.g. would we expose the associated constants and consider them fair game to return from third party renderers?

Reply:
Also, would it be ok for ReactDOM to reach into `schedulerPriorityToLanePriority` which is conceptually an internal? I guess we already do in the event system.